apps/system/lockscreen/js/lockscreen_inputpad.js

```diff
@@ -202,14 +220,18 @@
         }
 
         this.states.passCodeEntered += key;
-        this.updatePassCodeUI();
+        this.updatePassCodeUI('');
 
         if (this.states.padVibrationEnabled) {
           navigator.vibrate(this.configs.padVibrationDuration);
         }
 
         if (this.states.passCodeEntered.length === 4) {
-          this.lockScreen.checkPassCode(this.states.passCodeEntered);
+          (function(value) {
+            setTimeout(function() {
+              this.lockScreen.checkPassCode(value);
+            }, CODE_VALIDATION_TIMEOUT);
```

Comment:
And while I'm on the topic, if I tell you to do something that seems stupid, it might very well be something stupid, and you should feel free to debate it with me...